frontend/src/metabase/xray/components/InsightCard.jsx

```diff
@@ -0,0 +1,368 @@
+import React, { Component } from 'react'
+import {
+    //formatListOfItems,
+    formatTimeWithUnit,
+    //inflect
+} from "metabase/lib/formatting";
+import Icon from "metabase/components/Icon";
+import { Link } from "react-router";
+import Question from "metabase-lib/lib/Question";
+import { TermWithDefinition } from "metabase/components/TermWithDefinition";
+import { t } from 'c-3po'
+
+const InsightText = ({ children }) =>
+    <p className="text-paragraph">
+        {children}
+    </p>
+
+const Feedback = ({ insightType }) =>
+    <div className="flex align-center px1">
+        {t`Was this helpful?`}
+        <div className="ml-auto text-bold">
+            <a className="text-brand-hover" data-metabase-event={`InsightFeedback;${insightType};Yes`}>
+                {t`Yes`}
+            </a>
+            <a className="text-brand-hover ml1" data-metabase-event={`InsightFeedback;${insightType};No`}>
+                {t`No`}
+            </a>
+        </div>
+    </div>
+
+export class NormalRangeInsight extends Component {
+    static insightType = "normal-range"
+    static title = "Normal range of values"
+    static icon = "insight"
+
+    render() {
+        const { lower, upper, features: { model } } = this.props
+        return (
+            <InsightText>
+                Most of the values for { model.display_name || model.name } are between <b>{ lower }</b> and <b>{ upper }</b>.
+            </InsightText>
+        )
+    }
+}
+
+export class NilsInsight extends Component {
+    static insightType = "nils"
+    static title = "Missing data"
+    static icon = "warning"
+
+    render() {
+        const { quality, filter, features: { table } } = this.props
+
+        const viewAllRowsUrl = table && Question.create()
+            .query()
+            // imitate the required hydrated metadata format
+            .setTable({ ...table, database: { id: table.db_id }})
+            .addFilter(filter)
+            .question()
+            .getUrl()
+
+        // construct the question with filter
+        return (
+            <InsightText>
+                You have { quality } missing (null) values in your data.
+                <span> </span>
+                { table && <span><Link to={viewAllRowsUrl}>View all rows</Link> with missing value.</span> }
+            </InsightText>
+        )
+    }
+}
+
+export class ZerosInsight extends Component {
+    static insightType = "zeros"
+    static title = "Zeros in your data"
+    static icon = "warning"
+
+    render() {
+        const { quality, filter, features: { table } } = this.props
```

Comment:
@kdoh what's our policy on semicolons in Javascript? 

Personally I don't like implicit semicolons coming back to bite me in the ass so I've always preferred living in a land where we are explicit and have the linter prevent implicit ones. That way we don't make mistakes like the infamous

```javascript
return
{
    a: 100
}
```

Which JS interprets as

```javascript
return;
{
    a: 100
};
```

or 

```javascript
a = b + c
(d + e).print()
```

which gets interpreted as 

```javascript
a = b + c(d + e).print();
```

I know it's fun to be a JSHipster and pretend Javascript is just like python and doesn't need semicolons but the fact is sooner or later omitting them is going to introduce bugs. I think [this commenter on HN](https://news.ycombinator.com/item?id=2183842) says it better than I ever could:

> Omitting semicolons in JavaScript is like omitting braces from one line if blocks.
It's all well and good until someone turns this:
>
> ```c
> if (conditional)
>     statement;
> ````
>
> Into this without realizing what they've just done:
>
> ````c
> if (conditional)
>     statement;
>     statement2;
> ````
>
>Yes, you can skip semicolons. And yes it will work the vast majority of the time. But the amount of time you save will be eclipsed by that one time someone makes a change to the code that breaks in an unpredictable subtle way due to your purely aesthetic shorthand.
>Writing maintainable code is about learning humility. There's no need to show off your understanding of the deeper points of JavaScript semicolon insertion in your production code base. Just insert the semicolons, I promise you you will learn to not be distracted in no time.

Reply:
So we don't have a current lint rule or enforcement of this which I know is goofy and no good. I personally tend to be on team no semi-colons because I think the overall noise reduction makes it easier for me personally to write JS and I know the gotchas so I can avoid them, but I understand the arguments against it.

In my mind, reviewing code for style tends to get into personal tastes and philosophical debates I'd rather continue to not read on HN, so my real answer is that we should have robots do it for us. My vote would be to use https://prettier.io/ (or similar) for style rulse vs a linter so that team members and contributors can just write how they write and know it'll be formatted properly on the other side. Here's [more info](https://prettier.io/docs/en/comparison.html) on that.

My vote would be to use the time it would take to go add `;` everywhere or remove it, to set up said robot. We can add a lint rule to our current lint set up if it's really nagging at you, but it's inconsistent enough throughout the project that in either case we'd have to run some sort of ``--fix` which would create an enormous diff, so in my mind I prefer the solution that gives me more time to think about the what and not the formatting.

Reply:
the earlier post-commit hook configuration of Prettier made it cumbersome to use – I would love to have auto-formatting as part of my daily workflow but not sure how it would feel most natural